openff/interchange/smirnoff/_create.py

```diff
@@ -423,6 +423,43 @@ def _virtual_sites(
     interchange.collections.update({"VirtualSites": virtual_site_handler})
 
 
+def _build_collection(
+    collection_class: type[SMIRNOFFCollection],
+    handler_classes: list[type[ParameterHandler]],
+    interchange: Interchange,
+    force_field: ForceField,
+    topology: Topology,
+):
+    kwargs = {}
+
+    # Always attach handlers
+    if len(handler_classes) == 1:
+        kwargs["parameter_handler"] = force_field[handler_classes[0]._TAGNAME]
+    else:
+        kwargs["parameter_handler"] = [force_field[cls._TAGNAME] for cls in handler_classes]
+
+    kwargs["topology"] = topology
+
+    try:
+        from smirnoff_plugins.collections.vsites import _VsitePlugin  # type: ignore[import-not-found]
+    except ImportError:
+        _VsitePlugin = None
```

Comment:
Is there any way to avoid a dependency loop here?

Reply:
Maybe `hasattr(collection, "allowed_vdw_parameter_handlers")`?

Reply:
Either seems to work okay in terms of code, I think the attribute checking is a little cleaner but I can be talked out of it.

It's just that A <-> B packaging dependencies are something I try to avoid wherever possible, even though in terms of pure code the import-checking is a fine solution